settings/src/script.js

```diff
@@ -66,9 +71,19 @@ function Main() {
 	 */
 	function clickScreenLink( event, screen ) {
 		event.preventDefault();
+
+		// Reset to initial after navigating away from a page.
+		if ( userDataNeedsReset ) {
+			editUserData( userData );
+		}
+
 		setScreen( screen );
 	}
 
+	if ( ! userData ) {
```

Comment:
I believe I originally had `isResolving` here, but ran into an edge-case where it was `false` but `userData` was still unavailable. I believe that would've been due to the order of operations of likely retrieving the userData before the status.

I haven't verified if this is resolved by using `useEntityRecord()` instead, which handles those operations in the correct order (i assume). It also offers a `hasResolved` which could be used instead.

By using `! userData` here it does leave an endless-spinner on the page in the event of the userData being unable to be retrieved (where it would show a UI with "your email is undefined"), such as when you attempt to edit a user you can't edit.. but that's been handled in PHP now instead.

Reply:
The cause is that `isResolving` is only truthful once resolution has started, not before. So before it tried to fetch the record it wasn't resolved and it wasn't resolving.

Switching to `! hasResolved` works here, as would `status === SUCCESS`, but I've gone for the former.
This also allowed removing the `<Spinner />` from the child-classes, as it's safe to assume the parent has waited for the record to exist.

Reply:
That makes sense 👍🏻